### PR TITLE
[RELEASE] v1.1.2

### PR DESCRIPTION
**Pull request by CybotTM**
Release v1.1.2 on the TYPO3_12 maintenance branch. Consolidates the post-v1.1.1 work:

- **[#92](https://github.com/netresearch/t3x-nr-image-optimize/pull/92)** – #70 follow-up (error_log diagnostics on 400 responses, per-request allowed-roots cache, functional symlink test)
- **[#94](https://github.com/netresearch/t3x-nr-image-optimize/pull/94)** – CI coverage driver switched from pcov to xdebug
- **[#88](https://github.com/netresearch/t3x-nr-image-optimize/pull/88)** – filesystem-root-as-public-path fix (landed earlier, not yet in a tag)

## Who needs this

**Chemnitz** (TYPO3 v12.4.45 + AWS ECS/Fargate + EFS): the 400-response ran on every uncached image variant because `public/fileadmin`, `public/processed` and `public/uploads` are all symlinks to `/mnt/efs/cms/...`. The follow-up fix in this release makes the behavior diagnosable and closes the last latent gap (transient `StorageRepository` failures no longer poison the allowed-roots cache for the rest of the worker's life).

Addresses OPSCHEM-478.

## Changes

See CHANGELOG.md for the full entry. Top-level:

- **BUGFIX** – #70 follow-up (diagnosability + cache robustness), filesystem-root edge case (#88)
- **TESTS** – new `ProcessorSymlinkedFileadminTest` (functional, mirrors Chemnitz EFS layout), unit regression for cache non-poisoning
- **CI** – xdebug as coverage driver, functional tests now actually run in CI, Imagick locally and in CI

## Release procedure

After merge:

1. Create **signed** tag `v1.1.2` on the TYPO3_12 branch
2. Push tag
3. Create GitHub Release manually with `gh release create v1.1.2 --latest=false --title "v1.1.2" --notes "…"` — `--latest=false` is critical to preserve the main-branch v2.2.2 "Latest" badge
4. `publish-to-ter.yml` picks up `release:published` and publishes to TER

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,69 @@
...
+## BUGFIX
+
+- **#70 follow-up diagnosability & cache robustness.** The 1.1.0/1.1.1 fixes
+  silently returned HTTP 400 to clients when path validation rejected a
+  request, leaving admins no signal to diagnose from. `Processor::generateAndSend()`
+  now logs both 400 branches via `error_log()`: the URL-pattern-mismatch
+  path at info level (scanners hit this constantly, so warning would drown
+  out genuine issues) and the path-outside-allowed-roots path at warning
+  level with full diagnostic context (url, pathOriginal, pathVariant,
+  which check failed, allowedRoots, publicPath).
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply CHANGELOG.md:12]: Addressed in follow-up PR https://github.com/netresearch/t3x-nr-image-optimize/pull/103 (commit 22c99fb). The wording 'info' / 'warning' was inaccurate — `error_log()` has no level distinction; both 400 branches write at the same SAPI level. Reworded to describe what is actually different (short one-line vs full diagnostic context) and added an explicit note that error_log() has no severity tiers. The published v1.1.2 release notes are immutable so they still carry the original wording, but the in-tree CHANGELOG is now accurate.
